courtbot/states/CA/__init__.py

```diff
@@ -0,0 +1,123 @@
+
+import requests
+
+"""A package for retreiving court calendar data from Santa Clara County, California"""
+
+counties = [ 'santa clara' ]
+
+courts = [ 'santa clara' ]
+
+court_names = ["Civil",
+               "Probate",
+               "Family",
+               "Juvenile",
+               "Criminalhoj",
+               "CriminalPaloAlto",
+               "CriminalSouthCounty"]
+
+class Court:
+
+    def __init__(self, name, department):
+        self.__name = name
+        self.__dept = department
+
+    def setDepartment(self, dept_name):
+        self.__dept = dept_name
+
+    def getDepartment(self):
+        return self.__dept
+
+class CourtAppearance:
+
+    def __init__(self):
+        self.__when = ''
+
+    def __call__(self, appear_date, case_num):
+
+        cases = get_list( appear_date )
+
+        for case in cases:
+            if case['caseNbr'] == case_num:
+                return case
+
+        return {}
+
+    def setCourtName(self, name):
+        self.__court_name = name
+
+    def getCourtName(self):
+        return self.__court_name
+
+    court_name = property(getCourtName, setCourtName)
+
+    def setWhen(self, appear_datetime):
+        self.__when = appear_datetime
+
+    def getWhen(self):
+        return self.__when
+
+    when = property(getWhen, setWhen)
+
+    def setCaseNumber(self, case_number):
+        self.__case_number = case_number
+
+    def getCaseNumber(self):
+        return self.__case_number
+
+    case_number = property(getCaseNumber, setCaseNumber)
+
+    def setParties(self, parties):
+        self.__parties = parties
+
+    def getParties(self):
+        return self.__psrties
+
+    parties = property(getParties, setParties)
+
+    def setCourt(self, court_name, department):
+        self.__court = Court(name, department)
+
+    def getCourt(self):
+        return self.__court
+
+    court = property(getCourt, setCourt)
+
+get_case = CourtAppearance()
```

Comment:
Not sure about the `get_case` and `get_list` usage... It might be a stylistic thing. 

I see that these are instance variables created in the class.  And that - b/c you overwrite the `__call__` method, you can use it like a function (which `get_case()` works as a function).

But in general, it makes reading this code really confusing. 

Normally - programmatically, when you create instances in class definition, this types of set up normally is for when you want a singleton. That is - you only want - one instance of the class to be created.  This is needed in some cases.  But I think for this module there is no specific requirement for these to be singleton. 

The references between the classes (ie using `get_list` within the `__call__` method of `get_case`. - especially using these instance variables also makes it hard to read.  (And if these tracks states - it will be really confusing).  In this scenario, it is pretty simple to just create an instance of court apparencelist and call that method.

But.... why is CourtApparenaceList a class at all - as it does not have any internal variable to track?  the `__call__` function can be migrated to `CourtApparence` as a function in which then the `__call__` function (please rename it), can use. 

So my suggestion is that: 
```
class CourtAppearance:
   # replace: def __call__(self, appear_date, case_num): 
   def get_case(self, appear_date, case_num):
      cases = self.get_list( appear_date )
      ... 

   # pull in the def __call__ .. from CourtApparance
   def get_list(self, appear_date): 
       ... 

# usage in your test case. 
courtApp = court.CourtApparenance()
courtApp.get_case('2021-09-23', 'C1918672')
courtApp.get_list('2021-09-23')
```

Reply:
"why is CourtApparenaceList a class at all" - I think I agree it does not make a lot of sense. Again, I was trying to do something like something else....

This bit of code:
```
# usage in your test case. 
courtApp = court.CourtApparenance()
courtApp.get_case('2021-09-23', 'C1918672')
courtApp.get_list('2021-09-23')
```
makes a lot of sense. But here is a question. Is that how the interface to the OSCN project is written? Right now, the courtbot-library project is, for OK, just passing into the OSCN project. The people at Tulsa have said that they want courtbot-library to be a re-usable framework for other brigades to use.

Is the code as you have written it above compatible with what the OSCN project does. If it is not, how do we express the interface that all projects using the courtbot-library project should use?

I do not know python class design, as least how it is done here, well enough to put this into a suggestion for them.

Can you write a PR for the courtbot-library which suggests an interface that both the OK and CA projects should use?

Reply:
Let me see about that....